### PR TITLE
Add a generic publication script

### PR DESCRIPTION
**Pull request by wrouesnel**
This is a very minimal wrapper for `Publish_Subdirectory_To_Branch` which is designed to work with `setup-version` to provide a generic site publication solution.

### CODE REVIEW

File: ci-publish-site.bsh [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    exit 1
+fi
+
+if [ -z "$dir" ]; then 
+    echoerr "Directory cannot be blank."
+    exit 1
+fi
+
+if [ -z "$commit_msg" ]; then
+    echoerr "Commit Message cannot be blank"
```
Comment (stephanh) [inline ci-publish-site.bsh]: Couldn't we just use a default commit message?

File: ci-publish-site.bsh [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import lib-ci
+
+CI_Env_Adapt $(CI_Env_Get)
+
+branch="$1"
+dir=$(readlink -f $2)
+commit_msg="$3"
+
+if [ -z "$branch" ]; then
+    echoerr "Git branch not specified."
```
Comment (stephanh) [inline ci-publish-site.bsh]: Can we default to `gh-pages`?

File: tests/test-ci-publish-site.bsh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+# Check fail when empty VERSION
+touch VERSION
+WVFAIL ../ci-publish-site.bsh "gh-pages" "$TMPBASE/site" "Commit Msg"
+
+# Should pass when a version file exists
+echo "1.0.0" >> VERSION
+WVPASS ../ci-publish-site.bsh "gh-pages" "$TMPBASE/site" "Commit Msg"
+
+# Check the we end up with something sensible in the base directory
+WVPASS [ -e "$TMPBASE/site/_config.yml" ]
```
Comment (stephanh) [inline tests/test-ci-publish-site.bsh]: Probably redundant since the next test is a more thorough version of this check.

**Comment by stephanh** [comment]
:+1: Once you add defaults and the atexit PR is sorted.

**Comment by stephanh** [comment]
Why did you close this instead of merging?

File: ci-publish-site.bsh (hunk 1 above)
Comment (stephanh) [inline ci-publish-site.bsh]: You didn't add the defaults?

**Comment by wrouesnel** [comment]
I didn't? I see "merged". It popped to "closed" when I pushed.

I missed the defaults - I'll open a new PR tomorrow to add them.

**Pull request closed without merging**